Include React 18 as peer dependency

### Versions:

- `@inertiajs/inertia` version: 0.11.0
- `@inertiajs/inertia-react` version: 0.8.0

### Describe the problem:

<!--
  Explain the behavior you're seeing that you think is a bug,
  and explain how you think things should behave instead.
-->
Inertia works fine with React 18, which is why it should be included as a peer dependency.

Due to the missing peer dependency you get an error every time you do something in npm:

```bash
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: undefined@undefined
npm ERR! Found: react@18.0.0
npm ERR! node_modules/react
npm ERR!   dev react@"^18.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.9.0 || ^17.0.0" from @inertiajs/inertia-react@0.8.0
npm ERR! node_modules/@inertiajs/inertia-react
npm ERR!   dev @inertiajs/inertia-react@"^0.8.0" from the root project
```

--- a/packages/inertia-react/package.json
+++ b/packages/inertia-react/package.json
@@ -36,11 +36,11 @@
     "eslint-plugin-react": "^7.14.3",
     "eslint-plugin-react-hooks": "^4.0.0",
     "microbundle": "^0.12.0",
-    "react": "^16.9.0 || ^17.0.0"
+    "react": "^16.9.0 || ^17.0.0 || ^18.0.0"
   },
   "peerDependencies": {
     "@inertiajs/inertia": "^0.11.0",
-    "react": "^16.9.0 || ^17.0.0"
+    "react": "^16.9.0 || ^17.0.0 || ^18.0.0"
   },
   "dependencies": {
     "lodash.isequal": "^4.5.0"

